app/models/Photo.java

```diff
@@ -0,0 +1,193 @@
+    @JsonView(View.AdditionalDetails.class)
+    private String bucket;
+    @JsonView(View.AllDetails.class)
+    private String name;
+    @Transient
+    @JsonView(View.AdditionalDetails.class)
+    private File file;
+    @JsonView(View.AllDetails.class)
+    private boolean isDefault;
+    @JsonView(View.AllDetails.class)
+    @ManyToOne(cascade = CascadeType.ALL)
+    private Restaurant restaurant;
+
+    public Photo(Http.MultipartFormData.FilePart upload, int restaurantId) {
+        String fileExtension = FilenameUtils.getExtension(upload.getFilename());
+        name = UUID.randomUUID().toString() + "." + fileExtension;
+        file = upload.getFile();
+        this.restaurant = PersistenceManager.getRestaurantById(restaurantId);
+        isDefault = restaurant.getPhotos().size() == 0 ? true : false;
+    }
+
+	@JsonView(View.AllDetails.class)
+    public URL getUrl() throws MalformedURLException {
+        return new URL("https://s3.amazonaws.com/" + bucket + "/" + getName("original"));
+    }
+
+    @Override
+    public void delete() {
+        if (S3Plugin.amazonS3 == null) {
+            Logger.error("Could not delete because amazonS3 was null");
+            throw new RuntimeException("Could not delete");
+        }
+        else {
+            S3Plugin.amazonS3.deleteObject(bucket, getName("original"));
+            S3Plugin.amazonS3.deleteObject(bucket, getName("medium"));
+            S3Plugin.amazonS3.deleteObject(bucket, getName("thumbnail"));
+            super.delete();
+        }
+    }
+
+    @Override
+    public void save() {
+        if (S3Plugin.amazonS3 == null) {
+            Logger.error("Could not save because amazonS3 was null");
+            throw new RuntimeException("Could not save");
+        }
+        else {
+            this.bucket = S3Plugin.s3Bucket;
+            super.save(); // assigns an photoId
+
+            uploadPhotoToS3(bucket, getName("original"), this.file);
+            uploadPhotoToS3(bucket, getName("thumbnail"), getPhotoInOtherSize(50, 50));
+            uploadPhotoToS3(bucket, getName("medium"), getPhotoInOtherSize(500, 500));
+
+        }
+    }
+
+    private String getName(String size) {
+        return "photos/restaurants/" + restaurant.getRestaurantId() + "/" + photoId + "/" + size + "/" + name;
+    }
+
+    private void uploadPhotoToS3(String bucket, String name, File file){
+        PutObjectRequest putObjectRequest = new PutObjectRequest(bucket, name, file);
+        putObjectRequest.withCannedAcl(CannedAccessControlList.PublicRead); // public for all
+        S3Plugin.amazonS3.putObject(putObjectRequest); // upload file
+    }
+
+    private File getPhotoInOtherSize(int x, int y){
+        BufferedImage in = null;
+        File temp = new File(name);
+        try {
+            in = ImageIO.read(file);
+            if(in != null){
+                in = resize(in, x, y);
+            }
+            ImageIO.write(in, FilenameUtils.getExtension(name).toUpperCase(), temp);
```

Comment:
extension can be null, so this is not good approach (test it and see that you will get a NPE)